osu.Game.Rulesets.Mania/Beatmaps/Patterns/PatternGenerator.cs

```diff
@@ -12,6 +15,8 @@ namespace osu.Game.Rulesets.Mania.Beatmaps.Patterns
     /// </summary>
     internal abstract class PatternGenerator
     {
+        private const int max_rng_iterations = 100;
```

Comment:
on a first look this number feels quite arbitrarily high. at very least can we have a comment as to how you came up with this / how it was tested?

Reply:
Yeah it can probably be safely reduced to 20 or so, it's pretty arbitrary.